### PR TITLE
IGNITE-12361 Migrate Flume module to ignite-extensions

### PR DESCRIPTION
**Pull request by samaitra**

### CODE REVIEW

File: modules/flume-ext/config/example-ignite.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                            <list>
+                                <value>127.0.0.1:47500</value>
+                            </list>
+                        </property>
+                    </bean>
+                </property>
+            </bean>
+        </property>
+    </bean>
+</beans>
```
Comment (ezhuravl) [inline modules/flume-ext/config/example-ignite.xml]: Is this file needed at all? In PR https://github.com/apache/ignite/pull/7227/files#diff-3039d63ae06ffade7d91848a19d09fb3 I see that before that we had only one config file in flume:modules/flume-ext/src/test/resources/example-ignite.xml - can we use only it here?

File: modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    /** Cache name. */
+    private static final String CACHE_NAME = "testCache";
+
+    /**
+     * @throws Exception {@link Exception}.
+     */
+    @Test
+    public void testSink() throws Exception {
+        IgniteConfiguration cfg = loadConfiguration("example-ignite.xml");
```
Comment (ezhuravl) [inline modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: Can we use modules/flume-ext/src/test/resources/example-ignite.xml here ?

File: modules/flume-ext/config/example-ignite.xml (hunk 1 above)
Comment (samaitra) [reply modules/flume-ext/config/example-ignite.xml]: @ezhuravl This file is used when we are running the build inside local folder like `ignite-extensions/modules/flume-ext`. The IgniteSinkTest extends GridCommonAbstractTest and this has dependency based on GridTestProperties to look up for example-ignite.xml inside config folder.

File: modules/flume-ext/pom.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        <dependency>
+            <groupId>org.apache.ignite</groupId>
+            <artifactId>ignite-core</artifactId>
+            <version>${ignite-core.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.flume</groupId>
+            <artifactId>flume-ng-core</artifactId>
+            <version>${flume.ng.version}</version>
```
[the remark this answers is not in the document]
Comment (samaitra) [reply modules/flume-ext/pom.xml]: I had added `{flume.ng.version}` in parent/pom.xml

https://github.com/apache/ignite-extensions/blob/master/parent/pom.xml#L69

File: modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java (hunk 2 above)
Comment (samaitra) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @ezhuravl I had tested with `modules/flume-ext/src/test/resources/example-ignite.xml` and it fails the build as GridTestProperties lookup for configuration inside `modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml`.

I get the following error:

```
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: IOException parsing XML document from URL 
```

Comment (ezhuravl) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: Can you share a stacktrace?
The thing is that we have 2 identical config files, I don't see why do we need it here.

Comment (samaitra) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @ezhuravl please find the stacktrace below

[2020-03-21 12:29:31,525][INFO ][test-runner-#1%flume.IgniteSinkTest%][XmlBeanDefinitionReader] Loading XML bean definitions from URL [file:/Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml]
[2020-03-21 12:29:31,528][ERROR][main][root] Test failed.
class org.apache.ignite.IgniteCheckedException: Failed to instantiate Spring XML application context.
	at org.apache.ignite.testframework.junits.GridAbstractTest.loadConfiguration(GridAbstractTest.java:1474)
	at org.apache.ignite.stream.flume.IgniteSinkTest.testSink(IgniteSinkTest.java:54)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.apache.ignite.testframework.junits.GridAbstractTest$7.run(GridAbstractTest.java:2117)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: IOException parsing XML document from URL [file:/Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml]; nested exception is java.io.FileNotFoundException: /Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml (No such file or directory)
	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:344)
	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:304)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:181)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:217)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:188)
	at org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:252)
	at org.springframework.context.support.AbstractXmlApplicationContext.loadBeanDefinitions(AbstractXmlApplicationContext.java:127)
	at org.springframework.context.support.AbstractXmlApplicationContext.loadBeanDefinitions(AbstractXmlApplicationContext.java:93)
	at org.springframework.context.support.AbstractRefreshableApplicationContext.refreshBeanFactory(AbstractRefreshableApplicationContext.java:129)
	at org.springframework.context.support.AbstractApplicationContext.obtainFreshBeanFactory(AbstractApplicationContext.java:614)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:515)
	at org.springframework.context.support.FileSystemXmlApplicationContext.<init>(FileSystemXmlApplicationContext.java:140)
	at org.springframework.context.support.FileSystemXmlApplicationContext.<init>(FileSystemXmlApplicationContext.java:84)
	at org.apache.ignite.testframework.junits.GridAbstractTest.loadConfiguration(GridAbstractTest.java:1471)
	... 11 more
Caused by: java.io.FileNotFoundException: /Users/MY_HOME/git/ignite-extensions/modules/flume-ext/modules/flume-ext/src/test/resources/example-ignite.xml (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
	at org.springframework.core.io.UrlResource.getInputStream(UrlResource.java:169)
	at org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:330)
	... 24 more

Comment (ezhuravl) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: I removed config/example-ignite.xml from repository and didn't see any exceptions in this test. I think there might a problem with IGNITE_HOME or something like that on your side.

Comment (samaitra) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @ezhuravl I renamed the file ignite-extensions/config/example-ignite.xml to ignite-extensions/config/example-ignite1.xml and ran the build from folder of ignite-extensions.

` mvn package -Dmaven.javadoc.skip=true`

I see following results :


```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] ignite-flume-ext ................................... SUCCESS [ 21.010 s]
[INFO] ignite-pub-sub ..................................... FAILURE [  4.168 s]
[INFO] ignite-spring-boot-autoconfigure ................... SKIPPED
[INFO] ignite-client-spring-boot-autoconfigure ............ SKIPPED
[INFO] ignite-zeromq-ext .................................. SKIPPED
[INFO] apache-ignite-extensions ........................... SKIPPED
[INFO] ------------------------------------------------------------------------
```
I agree that ignite-extensions/config/example-ignite.xml is not used by flume-ext module but used by other modules.

I have reverted the changes in ignite-extensions/config/example-ignite.xml so that it is not part of this PR.

Comment (samaitra) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @ezhuravl I have also removed the ignite-extensions/modules/flume-ext/config/example-ignite.xml. I ran the build from ignite-extensions folder and also from ignite-extensions/modules/flume-ext folder and both time the build passed.

I also ran the tests in IDE and it is also passing.

Comment (ezhuravl) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @samaitra thanks, I was talking about ignite-extensions/modules/flume-ext/config/example-ignite.xml., since it's a part of this module.

Comment (samaitra) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @ezhuravl Thank you, yes I have removed the `ignite-extensions/modules/flume-ext/config/example-ignite.xml` file. Please let me know if any other changes required or if this PR looks good.

Comment (ezhuravl) [reply modules/flume-ext/src/test/java/org/apache/ignite/stream/flume/IgniteSinkTest.java]: @samaitra Yes, it looks good now and I think it can be merged
